sematic/api/tests/fixtures.py

```diff
@@ -74,25 +75,58 @@ def _request_callback(request):
 
 
 @contextlib.contextmanager
-def mock_user_settings(settings: Dict[user_settings.SettingsVar, Any]):
-    # Force load everything first
-    user_settings.get_all_user_settings()
+def _mock_settings_file(file_path: Optional[str]) -> Generator[Any, Any, None]:
 
-    original_settings = user_settings._settings
+    with tempfile.NamedTemporaryFile("w") as tf:
 
-    user_settings._settings = {
-        "default": {key.value: value for key, value in settings.items()}
-    }
+        if file_path is not None:
+            with open(file_path, "r") as f:
+                tf.write(f.read())
+                tf.flush()
 
-    try:
-        yield settings
-    finally:
-        user_settings._settings = original_settings
+        with mock.patch(
+            "sematic.user_settings._get_settings_file",
+            return_value=tf.name,
+            new_callable=mock.PropertyMock,
+        ):
+            # set up: invalidate the existing settings cache, if any
+            user_settings._settings = None
+
+            yield tf
+
+            # tear down: invalidate the mock settings cache
+            # it will be populated from the settings file on the next invocation
+            user_settings._settings = None
+
+
+@pytest.fixture(scope="function")
+def mock_settings_file(request: Any) -> Generator[Any, Any, None]:
```

Comment:
I don't follow how this fixture is intended to be used.

Reply:
Added docstring.